src/EnergyEfficientAgent.cpp

```diff
@@ -223,14 +221,15 @@ namespace geopm
         for (size_t ctl_idx = 0; ctl_idx < (size_t) m_num_freq_ctl_domain; ++ctl_idx) {
             struct m_region_info_s current_region_info {
                 .hash = (uint64_t)m_platform_io.sample(m_signal_idx[M_SIGNAL_REGION_HASH][ctl_idx]),
-                .hint = (uint64_t)m_platform_io.sample(m_signal_idx[M_SIGNAL_REGION_HINT][ctl_idx]),
+                .hint = GEOPM_REGION_HINT_UNKNOWN,
                 .runtime = m_platform_io.sample(m_signal_idx[M_SIGNAL_REGION_RUNTIME][ctl_idx]),
                 .count = (uint64_t)m_platform_io.sample(m_signal_idx[M_SIGNAL_REGION_COUNT][ctl_idx])};
             // If region hash has changed, or region count changed for the same region
             // update current region (entry)
             if (m_last_region_info[ctl_idx].hash != current_region_info.hash ||
                 m_last_region_info[ctl_idx].count != current_region_info.count) {
-                if (do_learning(current_region_info.hash, current_region_info.hint)) {
+                if ((current_region_info.hash != GEOPM_REGION_HASH_UNMARKED &&
```

Comment:
is this special case for unmarked an optimization?  I don't think this check is needed.

Reply:
This hash check was preserved from the deleted do_learning.  I kept it because otherwise this would be a behavior change as we don't want to learn for the unmarked buckets of time

Reply:
Okay thanks, makes sense.  When I suggested getting rid of the `do_learning()` interface, I did not realize that the check for unmarked was included in the function in addition to the find() result.  A different option to the do_learning() API would be  to have two sets of hashes one with the list of all low frequency regions, and another for all regions where learning should not occur.  The second set would contain the first.  I think this would be clearer when reading the code.  That is to say there would be two sets of uint64_t as members:
```
m_low_freq_hash = ApplicationSampler::region_hash_network();
m_no_learning_hash = m_low_freq_hash;
m_no_learning_hash.insert(GEOPM_REGION_HASH_UNMARKED);
```